Sparkle/SUUpdater.m

```diff
@@ -137,6 +118,43 @@ - (instancetype)initForBundle:(NSBundle *)bundle
     return self;
 }
 
+-(void)showModalAlertText:(NSString *)text informativeText:(NSString *)informativeText {
+    [self notifyWillShowModalAlert];
+    NSAlert *alert = [[NSAlert alloc] init];
+    alert.messageText = text;
+    alert.informativeText = informativeText;
+    [alert runModal];
+    [self notifyDidShowModalAlert];
+}
+
+-(void)checkIfConfiguredProperly {
+    BOOL hasPublicDSAKey = [self.host publicDSAKey] != nil;
+    BOOL isMainBundle = [self.host.bundle isEqualTo:[NSBundle mainBundle]];
+    BOOL hostIsCodeSigned = [SUCodeSigningVerifier hostApplicationIsCodeSigned];
+    NSURL *feedURL = [self feedURL];
+    BOOL servingOverHttps = [[[feedURL scheme] lowercaseString] isEqualToString:@"https"];
+    if (!isMainBundle && !hasPublicDSAKey) {
+        [self showModalAlertText:@"Insecure update error!"
+                 informativeText:@"For security reasons, you need to sign your updates with a DSA key. See Sparkle's documentation for more information."];
+    } else if (isMainBundle && !(hasPublicDSAKey || hostIsCodeSigned)) {
+        [self showModalAlertText:@"Insecure update error!"
+                 informativeText:@"For security reasons, you need to code sign your application or sign your updates with a DSA key. See Sparkle's documentation for more information."];
+    } else if (isMainBundle && !hasPublicDSAKey && !servingOverHttps) {
+        SULog(@"WARNING: Serving updates over HTTP without signing them with a DSA key is deprecated and may not be possible in a future release. Please serve your updates over https, or sign them with a DSA key, or do both. See Sparkle's documentation for more information.");
+    }
+
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101100
+    BOOL atsExceptionsExist = nil != [self.host objectForInfoDictionaryKey:@"NSAppTransportSecurity"];
```

Comment:
Very good idea!
However, having `NSAppTransportSecurity` in the `info.plist` is not sufficient guarantee that exceptions are in place.
Meaning - it might be empty or include other settings (thus update will still fail), but the alert won't show.
Consider checking if either `AllowArbitraryLoads` or `NSExceptionDomains` exists to increase chances the analysis is correct (unfortunately I can't think of a bullet-proof check - maybe someone else has a better idea).